### ISSUE TITLE
[Queue dashboard] Queues always show 100% error rate even with no errors

### ISSUE THREAD

**Issue by Fwang36** [getsentry/sentry-python#3420]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

1. [Follow the docs](https://docs.sentry.io/platforms/python/tracing/instrumentation/custom-instrumentation/queues-module/) to create a queue monitoring transaction
2. Open up the Queue monitoring dashboard and see 100% error rate.

![Image](https://github.com/user-attachments/assets/fed2fee2-7a4a-4490-9ee9-2713cabf0b8c)

3. See that transaction has no errors

![Image](https://github.com/user-attachments/assets/87e10e95-0de2-42b2-966b-573ad3a69b3d)

[Link to my Queue dashboard](https://francisorg.sentry.io/performance/trace/461aed25c5ed4cc7b0ae635760b60b13/?destination=messages&eventId=c23c7173b3fa4229a6b4e845e79e45a1&node=span-a46a13c5b5ba216c&node=txn-c23c7173b3fa4229a6b4e845e79e45a1&project=4504091267497984&source=queues_module&span.op=queue.publish&statsPeriod=14d&timestamp=1722951895&transaction=queue_producer_transaction)

### Expected Result

Error rate should not be 100%

### Actual Result

Error rate is 100%

### Product Area

Insights

**Comment by szokeasaurusrex** [comment]
@Fwang36 @edwardgou-sentry I played around with this a bit, and I believe what @Fwang36 is observing is expected behavior (although, I was unable to access the Queue Dashboard linked in the issue, so this is based somewhat on my inference of what you are trying to do).

Specifically, the problem seems to be that you are not setting the status to `"ok"` on the transaction. It looks like you already noticed this, but then attempted to solve the problem with `set_context`. In fact, all you need to do is to set the transaction's status to `ok` with `set_status`. You can also set the status in the `status` keyword argument when calling `start_transaction`.

In order for the trace to have an `ok` status, all transactions and spans in the trace need to have an `ok` status. If you did not set this status on the transactions, it is expected that the trace would have a non-`ok` status.

The following minimal example works as expected.

```python
import sentry_sdk
import time

def main():
    sentry_sdk.init(traces_sample_rate=1.0, debug=True)

    # Producer
    print("simulating producer")
    with sentry_sdk.start_transaction(op="test_op", name="publish") as tx:
        tx.set_status("ok")  # <------------------------------------------------- The missing piece -----------
        with sentry_sdk.start_span(op="queue.publish", description="Publishing a message") as span:
            span.set_status("ok")
            span.set_data("messaging.destination.name", "my_queue2")

            headers = {
                "sentry-trace": sentry_sdk.get_traceparent(),
                "baggage": sentry_sdk.get_baggage(),
            }

            message = {"headers": headers, "body": "Hello, World!"}

            print("pretend the message was published")

    time.sleep(0.1)

    # Consumer
    print("simulating consumer")
    tx = sentry_sdk.continue_trace(headers, op="test_op", name="consume")
    with sentry_sdk.start_transaction(tx):
        tx.set_status("ok")  # <------------------------------------------------- The other missing piece -----
        with sentry_sdk.start_span(op="queue.process", description="Consuming a message") as span:
            span.set_status("ok")
            span.set_data("messaging.destination.name", "my_queue2")

            headers = {
                "sentry-trace": sentry_sdk.get_traceparent(),
                "baggage": sentry_sdk.get_baggage(),
            }

            message = {"headers": headers, "body": "Hello, World!"}

            print("pretend the message was consumed")
            print("message: ", message["body"])


if __name__ == "__main__":
    main()
```

**Comment by szokeasaurusrex** [comment]
Let me know whether this fixes your issue. If it does, please transfer to the Docs repo, since we should likely specify in the docs page you referenced that the transaction and span status needs to be set to `ok` for the queues to be labelled as being error-free

**Comment by edwardgou-sentry** [comment]
Thanks for the follow up @szokeasaurusrex !
For the unknown status, the ui considers it an error because we followed the definition from the contexts interface docs page:
https://develop.sentry.dev/sdk/event-payloads/contexts/#trace-context
![Image](https://github.com/user-attachments/assets/485fb53e-bb04-4086-a02d-093bc15420fa)

If this is wrong, we can certainly change the ui behavior!

**Comment by Fwang36** [comment]
@szokeasaurusrex Yes, that works! thank you. calling `set_status("ok")` on the transaction marks the trace as ok.  I'll move this over to the docs repo so that can be added.

@edwardgou-sentry for the unknown status, I believe the transaction Performance Metric `failure_rate()` counts a transaction with an unknown status as a success based on these docs https://docs.sentry.io/product/performance/metrics/#failure-rate.  So I think there's precedent where unknowns dont count as errors
